### PR TITLE
Adding pin Rotation

### PR DESCRIPTION
**Pull request by deextercz**
I think it is not possible to create this feature for UWP.
The only way it to rotate images, but please let me know I am correct.

### CODE REVIEW

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/Logics/PinLogic.cs [hunk 1]
```diff
@@ -101,7 +102,7 @@ void OnInfoWindowClick(object sender, GoogleMap.InfoWindowClickEventArgs e)
             // lookup pin
             var targetPin = LookupPin(e.Marker);
 
-            // only consider event handled if a handler is present. 
+            // only consider event handled if a handler is present.
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/Logics/PinLogic.cs]: Please don't contain "not the purpose" changes.
Code cleaning is "not purpose" for this PR.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/Logics/PinLogic.cs [hunk 2]
```diff
@@ -242,10 +243,10 @@ protected override void OnUpdateType(Pin outerItem, Marker nativeItem)
 
         protected override void OnUpdateIcon(Pin outerItem, Marker nativeItem)
         {
-            if (outerItem.Icon != null && outerItem.Icon.Type == BitmapDescriptorType.View) 
+            if (outerItem.Icon != null && outerItem.Icon.Type == BitmapDescriptorType.View)
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/Logics/PinLogic.cs]: Please don't contain "not the purpose" changes.
Code cleaning is "not purpose" for this PR.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/Logics/PinLogic.cs [hunk 3]
```diff
@@ -242,10 +243,10 @@ protected override void OnUpdateType(Pin outerItem, Marker nativeItem)
...
             {
                 // If the pin has an IconView set this method will convert it into an icon for the marker
-                TransformXamarinViewToAndroidBitmap(outerItem, nativeItem);       
+                TransformXamarinViewToAndroidBitmap(outerItem, nativeItem);
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.Android/Logics/PinLogic.cs]: Please don't contain "not the purpose" changes.
Code cleaning is "not purpose" for this PR.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.iOS/Logics/PinLogic.cs [hunk 4]
```diff
@@ -106,7 +107,7 @@ void OnInfoTapped(object sender, GMSMarkerEventEventArgs e)
             // lookup pin
             var targetPin = LookupPin(e.Marker);
 
-            // only consider event handled if a handler is present. 
+            // only consider event handled if a handler is present.
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.iOS/Logics/PinLogic.cs]: Please don't contain "not the purpose" changes.
Code cleaning is "not purpose" for this PR.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps/Pin.cs [hunk 5]
```diff
@@ -16,6 +16,8 @@ public sealed class Pin : BindableObject
 
         public static readonly BindableProperty IsDraggableProperty = BindableProperty.Create("IsDraggable", typeof(bool), typeof(Pin), false);
 
+        public static readonly BindableProperty RotationProperty = BindableProperty.Create("Rotation", typeof(float), typeof(Pin), 0);
```
Comment (amay077) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps/Pin.cs]: I have error 

>System.TypeInitializationException: The type initializer for 'Xamarin.Forms.GoogleMaps.Pin' threw an exception. ---> System.ArgumentException: Default value did not match return type
Parameter name: defaultValue

Please change last parameter ``0`` to ``0f``.

``BindableProperty.Create("Rotation", typeof(float), typeof(Pin), 0f);``

**Comment by deextercz** [comment]
Sample added is it ok fornext review?

**Pull request merged**